File: c#/Query1.cs
using System;
using System.Collections.Generic;
using System.Threading;

namespace queries{
    class Query1{
        static List<List<string>> cantones;
        static List<string> seleccionarCanton(){
            Random random = new Random();
            int posCanton = random.Next(0, cantones.Count);
            List<string> canton = cantones[posCanton];
            cantones.RemoveAt(posCanton);
            return canton;
        }

        static void hiloEntregablesCanton(){
            List<string> canton = seleccionarCanton();
            // Se piden los entregables del cantón
            List<List<string>> entregables = new AccesoBaseDatos().consultaBaseDatos("EntregablesCanton " + canton[0]);
            string resultado = "----- " + canton[1] + " -----\n";
            for (int i = 0; i < entregables.Count; i++){
                resultado += (
                    "  Descripción: " + entregables[i][0] + "\n" +
                    "  Fecha de cumplimiento: " + entregables[i][1] + "\n" +
                    "  Valor de kpi: " + entregables[i][2] + "\n" +
                    "  Unidad de kpi: " + entregables[i][3] + "\n"
                );
            }
            resultado += "\n---------------";
            Console.WriteLine(resultado);
            new Temporizador().finalizarHilo();
        }

        public void ejecutar(){
            new Temporizador().IniciarTemporizador();
            // Se piden los cantones por procedimiento almacenado
            cantones = new AccesoBaseDatos().consultaBaseDatos("ObtenerCantones");
            for (int i = 0; i < 10; i++){
                // Se ejecutan los hilos
                Thread hilo = new Thread(hiloEntregablesCanton);
                // Se inicia el hilo
                hilo.Start();
            }
        }
    }
}

File: java/Query1.java
import java.util.List;
import java.lang.Math;

class HiloEntregablesCanton extends Thread{
    int idCanton = 0;
    String nombreCanton = "";

    public HiloEntregablesCanton(int idCanton, String nombreCanton){
        this.idCanton = idCanton;
        this.nombreCanton = nombreCanton;
    }

    public void run(){
        // Se llama al procedimiento almacenado que lista los entregables de un cantón
        List<List<String>> entregables = new AccesoBaseDatos().consultaBaseDatos("EntregablesCanton " + idCanton);
        String resultado = "----- "+nombreCanton+" -----\n";
        for(int i=0; i<entregables.size(); i++){
            resultado += (
                "  Descripción: "+entregables.get(i).get(0) + "\n" +
                "  Fecha de cumplimiento: "+entregables.get(i).get(1) + "\n" +
                "  Valor de kpi: "+entregables.get(i).get(2) + "\n" +
                "  Unidad de kpi: "+entregables.get(i).get(3) + "\n"
            );
        }
        resultado += "\n---------------";
        System.out.println(resultado);
        new Temporizador().finalizarHilo();
    }
}

public class Query1{
    static int generarNumeroAleatorio(int min, int max) {
        return (int) ((Math.random() * (max - min)) + min);
    }

    static void crearHilos(){
        new Temporizador().iniciarTemporizador();
        // Se llama al procedimiento almacenado para obtener los cantones
        List<List<String>> cantones = new AccesoBaseDatos().consultaBaseDatos("ObtenerCantones");
        for(int i=0; i<10; i++){
            int posCanton = generarNumeroAleatorio(0, cantones.size());
            // Se crean los hilos
            HiloEntregablesCanton hilo = new HiloEntregablesCanton(
                Integer.parseInt(cantones.get(posCanton).get(0)),
                cantones.get(posCanton).get(1)
            );
            hilo.start();
            cantones.remove(posCanton);
        }
    }

    public static void main(String[] args) {
        crearHilos();
    }
}